clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/ActiveActiveStoreIngestionTask.java

```diff
@@ -242,28 +253,43 @@ private byte[] prependReplicationMetadataBytesWithValueSchemaId(ByteBuffer metad
    * @param subPartition The partition to fetch the replication metadata from storage engine
    * @return The object containing RMD and value schema id. If nothing is found, return {@code Optional.empty()}
    */
-  private Optional<RmdWithValueSchemaId> getReplicationMetadataAndSchemaId(
+  Optional<RmdWithValueSchemaId> getReplicationMetadataAndSchemaId(
       PartitionConsumptionState partitionConsumptionState,
       byte[] key,
       int subPartition) {
     PartitionConsumptionState.TransientRecord cachedRecord = partitionConsumptionState.getTransientRecord(key);
     if (cachedRecord != null) {
-      hostLevelIngestionStats.recordIngestionReplicationMetadataCacheHitCount();
+      getHostLevelIngestionStats().recordIngestionReplicationMetadataCacheHitCount();
       return Optional.of(
           new RmdWithValueSchemaId(
               cachedRecord.getValueSchemaId(),
-              rmdProtocolVersionID,
+              getRmdProtocolVersionID(),
               cachedRecord.getReplicationMetadataRecord()));
     }
 
     final long lookupStartTimeInNS = System.nanoTime();
-    byte[] replicationMetadataWithValueSchemaBytes = storageEngine.getReplicationMetadata(subPartition, key);
-    hostLevelIngestionStats
+
+    ByteBuffer rmdWithValueSchemaByteBuffer = getRmdWithValueSchemaByteBufferFromStorage(subPartition, key);
+    byte[] replicationMetadataWithValueSchemaBytes =
+        rmdWithValueSchemaByteBuffer == null ? null : rmdWithValueSchemaByteBuffer.array();
+
+    getHostLevelIngestionStats()
         .recordIngestionReplicationMetadataLookUpLatency(LatencyUtils.getLatencyInMS(lookupStartTimeInNS));
-    if (replicationMetadataWithValueSchemaBytes == null) {
+    if (rmdWithValueSchemaByteBuffer == null) {
       return Optional.empty(); // No RMD for this key
     }
-    return Optional.of(rmdSerDe.deserializeValueSchemaIdPrependedRmdBytes(replicationMetadataWithValueSchemaBytes));
+    return Optional
+        .of(getRmdSerDe().deserializeValueSchemaIdPrependedRmdBytes(replicationMetadataWithValueSchemaBytes));
+  }
+
+  ByteBuffer getRmdWithValueSchemaByteBufferFromStorage(int subPartition, byte[] key) {
+    ValueRecord result =
+        SingleGetChunkingAdapter.getReplicationMetadata(getStorageEngine(), subPartition, key, isChunked(), null);
```

Comment:
Ehh, no. 
So my core understanding of RMD chunking is this feature will only be there when value chunking is enabled. I don't see a case where RMD alone can grow beyond the size limit without value growing beyond first (it's possible that we override the value with a smaller size new payload, but that would happen after value growing first.), even this could happen, we can just enable value chunking, but I still don't think this is possible.


Here the isChunked() is used to determined if key needs the chunking suffix for question. Please feel free to take a look at SingleGetChunkingAdapter.getReplicationMetadata(). Whether it is RMD chunking or not is not determined here, as when we fetch a manifest we will know this RMD is chunked.

Reply:
Maybe if you think this is causing confusion to reader, I can rename the method to isValueChunkingEnabled() or something similar.

Reply:
This is for ramping. As far as I can remember I did one scan and there is a use case with AA + chunking, that's why I introduced the flag for ramping. After the ramp, we should default this feature to True, and remove the unnecessary flag.

Reply:
I see. so that use case is using chunking without write compute, right?
For that specific use case, I assume the key stored in RMD CF is using chunking suffix, right?
But right now, this function is passing `isChunked()` to this lookup function call, which means the lookup will append the chunking suffix to the raw key bytes, won't it break this use case in prod?
To handle this specific use case and ramping purpose, I think we should use the `storeVersion.rmdChunkingEnabled` to decide whether the key used for RMD CF lookup should append the suffix or not, right?

Reply:
Oh looks like you are correct... hmm let me see....